src/account-info/AccountDetailsScreen.js

```diff
@@ -29,29 +36,51 @@ class AccountDetailsScreen extends PureComponent<Props> {
   };
 
   render() {
-    const { user } = this.props;
+    const { user, isDeactivated } = this.props;
     const title = {
       text: '{_}',
       values: {
         // This causes the name not to get translated.
         _: user.full_name || ' ',
       },
     };
-
-    return (
-      <Screen title={title}>
-        <AccountDetails user={user} />
+    let deactivatedText = null;
+    let privateMessageButton = null;
+    if (isDeactivated) {
+      deactivatedText = (
+        <RawLabel style={styles.deactivatedText} text="(This user has been deactivated)" />
+      );
+      privateMessageButton = (
+        <ZulipButton
+          style={styles.pmButton}
+          text="View private messages"
+          onPress={this.handleChatPress}
+          Icon={IconPrivateChat}
+        />
+      );
+    } else {
+      privateMessageButton = (
         <ZulipButton
           style={styles.pmButton}
           text="Send private message"
           onPress={this.handleChatPress}
           Icon={IconPrivateChat}
         />
+      );
+    }
+
+    return (
+      <Screen title={title}>
+        <AccountDetails user={user} />
+        {deactivatedText}
+        {privateMessageButton}
       </Screen>
     );
   }
 }
 
 export default connectFlowFixMe((state: GlobalState, props) => ({
   user: getAccountDetailsUserForEmail(state, props.navigation.state.params.email),
+  isDeactivated:
```

Comment:
Can you extract this to a selector? or better move this logic to `getAccountDetailsUserForEmail `, which by name says account details for email (and deactivated information belongs to this).

Reply:
Done